activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -142,11 +142,6 @@ def supports_insert_on_duplicate_update?
         true
       end
 
-      def field_ordered_value(column, values) # :nodoc:
-        field = Arel::Nodes::NamedFunction.new("FIELD", [column, values.reverse.map { |value| Arel::Nodes.build_quoted(value) }])
-        Arel::Nodes::Descending.new(field)
-      end
```

Comment:
I had assumed there would be a performance difference between `ORDER BY FIELD` and `ORDER BY CASE`, hence the specialization.  Do you know if that's true?

If so, we could conditionally call `super` instead:

```ruby
if values.include?(nil)
  super
else
  # fast path
end
```

Reply:
I tested with a table with 100k records and selecting the different number of records (up to 100k) and got basically the same timings.

Reply:
@fatkodima would you mind to share both SQL queries? I can check on real MySQL DB and also compare plans.

Reply:
```ruby
create_table :test do |t|
  t.string :value
end

class Test < ApplicationRecord
  self.table_name = :test
end

tests = 100_000.times.map { { value: ["one", "two", "three"].sample }}; nil
Test.insert_all tests

Test.in_order_of(:value, ["one", "two", "three"]).limit(100).length
```

Before
```sql
SELECT `test`.* FROM `test` WHERE `test`.`value` IN ('one', 'two', 'three') 
ORDER BY FIELD(`test`.`value`, 'three', 'two', 'one') DESC LIMIT 100
```

After
```sql
SELECT `test`.* FROM `test` WHERE `test`.`value` IN ('one', 'two', 'three') 
ORDER BY CASE WHEN `test`.`value` = 'one' THEN 1 WHEN `test`.`value` = 'two' THEN 2 
WHEN `test`.`value` = 'three' THEN 3 ELSE 4 END ASC LIMIT 100
```